src/app/components/media/MediaTimeline.js

```diff
@@ -0,0 +1,469 @@
+      });
+    });
+
+    tags
+      .filter(({ node }) => !!node)
+      .forEach(({
+        node,
+        node: {
+          id: instance,
+          fragment,
+          parsed_fragment: { t: [start_seconds, end_seconds] = [] } = {},
+          tag_text_object,
+        },
+      }) => {
+        if (!tag_text_object) return;
+
+        const { id, text: name } = tag_text_object;
+        if (!fragment) {
+          projecttags.push({ id, name });
+          return;
+        }
+
+        if (!entities[id]) {
+          entities[id] = {
+            id, name, project_tag: { id, name }, instances: [], node,
+          };
+        }
+
+        entities[id].type = 'tag';
+        entities[id].instances.push({ start_seconds, end_seconds, id: instance });
+      });
+
+    locations.filter(({ node: { id } }) => !!id).forEach(({
+      node, node: { id, parsed_fragment: { t: [start_seconds, end_seconds] }, content },
+    }) => {
+      const {
+        geolocation_viewport: { viewport, zoom = 10 },
+        geolocation_location,
+      } = JSON.parse(content).reduce((acc, { field_name, value_json }) =>
+        ({ ...acc, [field_name]: value_json }), {});
+
+      const {
+        properties: { name = id.trim() } = {},
+        geometry: { type = 'Point', coordinates = [0, 0] } = {},
+      } = geolocation_location;
+      const [lng, lat] = coordinates;
+
+      if (!entities[`place-${name}`]) {
+        entities[`place-${name}`] = {
+          id: `place-${name}`,
+          name,
+          viewport,
+          zoom,
+          lat,
+          lng,
+          type: type === 'Point' ? 'marker' : 'polygon',
+          test: { geolocation_location },
+          project_place: { id: `place-${name}`, name },
+          instances: [],
+          node,
+        };
+
+        if (type === 'Polygon') {
+          entities[`place-${name}`].polygon = coordinates[0].map(([lng, lat]) => ({ lat, lng }));
+        }
+      }
+
+      entities[`place-${name}`].instances.push({ id, start_seconds, end_seconds });
+    });
+
+    const data = {
+      project: {
+        projectclips: projecttags,
+        projecttags,
+        projectplaces: projecttags,
+      },
+      commentThreads,
+      videoClips: Object.values(entities).filter(({ project_clip }) => !!project_clip),
+      videoTags: Object.values(entities).filter(({ project_tag }) => !!project_tag),
+      videoPlaces: Object.values(entities).filter(({ project_place }) => !!project_place),
+      user: {
+        first_name,
+        id: userId,
+        last_name: '',
+        profile_img_url,
+      },
+    };
+
+    console.log({ data });
+
+    return {
+      duration, time, data, mediaId, dbid,
+    };
+  }
+
+  componentDidUpdate() {
+    const { fragment: { id } = {} } = this.props;
+    const instance = id ? document.querySelector(`*[data-instance-id="clip-${id.trim()}"]`) : null;
```

Comment:
We don't have refs to things inside the timeline from check-ui, we have data attributes that we can find in the DOM

Reply:
Then please change check-ui to use refs and props.

Reply:
I’m worried that this may end up being a larger update to **check-ui**. Is it reasonable to file it as a new Github issue or a Mantis ticket and tackle it outside of this PR?

Reply:
Makes total sense to me. Safest course of action would be to delete this `componentDidUpdate()` and add the feature later.

Reply:
fixed